managed/services/platform/platform.go

```diff
@@ -130,44 +137,28 @@ func (s *Service) Disconnect(ctx context.Context, req *platformpb.DisconnectRequ
 		return nil, err
 	}
 
-	err = models.DeletePerconaSSODetails(s.db.Querier)
-	if err != nil {
-		s.l.Errorf("Failed to delete SSO details: %s", err)
-		if e := s.UpdateSupervisordConfigurations(ctx); e != nil {
-			s.l.Errorf("%s %s", rollbackFailed, e)
-		}
-		return nil, err
-	}
-
 	userAccessToken, err := s.grafanaClient.GetCurrentUserAccessToken(ctx)
 	if err != nil {
 		if errors.Is(err, grafana.ErrFailedToGetToken) {
-			return nil, status.Error(codes.FailedPrecondition, "Failed to get access token. Please sign in using your Percona Account.")
+			return nil, errGrafanaAccessTokenFailed
 		}
 		s.l.Errorf("Disconnect to Platform request failed: %s", err)
 		return nil, err
 	}
 
 	err = s.client.Disconnect(ctx, userAccessToken, settings.PMMServerID)
-	needRecover := err != nil && !req.Force
```

Comment:
@oksana-grishchenko can we deprecate this flag now? or do we still need it?

Reply:
Looks like we don't need it anymore, this flag was introduced to prevent inserting the sso data back to the DB after a failed disconnect. 
Do you know if there was any specific reason why the functions calls were in that order initially? It looks like there is some story behind that order.

Reply:
@oksana-grishchenko i checked the spec https://confluence.percona.com/display/PMM/Single+Sign-On+-+PMM+Server+integration+with+Portal+and+Okta  and it mentions the exact same flow for Disconnect that this PR implements it seems it was implemented the wrong way initially.